### PR TITLE
example: disable gnrc modules without netif

### PR DESCRIPTION
**Pull request by OlegHahm**
Trying to find a solution to compile gnrc modules only if a network interface is available. I know that this solution is not very pretty, but it works and is the less intrusive and less ugly I could think of. Anyone with a better idea?

### CODE REVIEW

**Comment by authmillenon** [comment]
`PROVIDED_FEATURE`?

**Comment by authmillenon** [comment]
(what I meant was: only add `gnrc_netif_default` etc. if the board provides the feature)

**Comment by OlegHahm** [comment]
I thought about this, but a) the network interface is not necessarily provided by the board and b) one still has to check somewhere else in the build system if gnrc modules are required to build.

**Comment by authmillenon** [comment]
Okay, this fix is desperately needed, so let's see what Travis is doing.

**Comment by authmillenon** [comment]
Now all boards that DO provide a netif are failing :/

File: examples/default/Makefile [hunk 1]
```diff
@@ -67,4 +67,16 @@ ifneq (,$(filter fox,$(BOARD)))
 	USEMODULE += lsm303dlhc
 endif
 
+include $(RIOTBASE)/Makefile.dep
```
Comment (authmillenon) [inline examples/default/Makefile]: This would conflict with 3709

**Comment by authmillenon** [comment]
> a) the network interface is not necessarily provided by the board

Wasn't that the idea behind `FEATURES_PROVIDED`, that you can add to it if you put stuff on your board?

**Comment by authmillenon** [comment]
See https://github.com/OlegHahm/RIOT/pull/8

**Comment by OlegHahm** [comment]
> Now all boards that DO provide a netif are failing :/

I don't know why. If I include `Makefile.include` twice instead of `Makefile.dep` it's doing it correctly, but I get other errors (because of double definition).

**Comment by OlegHahm** [comment]
Ah, because the build system doesn't know the board folder yet.

**Comment by OlegHahm** [comment]
After thinking about the problem some more I think the solution is as simple as unsophisticated. This application is very special case. Usually a user either needs networking or she/he does not. So, there is no real need for the build system to build the networking features conditionally. Instead we can just make the decision at the one single place, where it's currently needed: in this example.

**Comment by authmillenon** [comment]
I don't really like it, but have no better idea. ACK, please squash.

**Comment by OlegHahm** [comment]
squashed

**Pull request merged**